src/ansys/pre_commit_hooks/add_license_headers.py

```diff
@@ -407,26 +445,49 @@ def find_files_missing_header() -> int:
         "git_repo": git_repo,
     }
 
+    # Run REUSE on root of the repository
+    git_root = values["git_repo"].git.rev_parse("--show-toplevel")
+
+    # git_root with correct line separators for operating system
+    os_git_root = git_root.replace("/", os.sep)
+
+    # Dictionary containing the asset folder information
+    assets = {
+        ".reuse": {
+            "path": os.path.join(".reuse", "templates"),
+            "default_file": f"{DEFAULT_TEMPLATE}.jinja2",
+        },
+        "LICENSES": {
+            "path": "LICENSES",
+            "default_file": f"{DEFAULT_LICENSE}.txt",
+        },
+    }
+
     # Add header arguments to parser. Arguments are: copyright, license, contributor,
     # year, style, copyright-style, template, exclude-year, merge-copyrights, single-line,
     # multi-line, explicit-license, force-dot-license, recursive, no-replace,
     # skip-unrecognized, and skip-existing
     header.add_arguments(parser)
 
-    # Run REUSE on root of the repository
-    git_root = values["git_repo"].git.rev_parse("--show-toplevel")
-
-    # Copy .reuse folder and LICENSES folder (if licenses are being checked)
-    copy_assets(git_root, args)
+    # Unlink default files & remove .reuse and LICENSES folders if empty
+    link_assets(assets, os_git_root, args)
 
+    # Project to run `REUSE <https://reuse.software/>`_ on
     proj = project.Project(git_root)
 
     # Get files missing headers (copyright and/or license information)
     missing_headers = list(list_noncompliant_files(args, proj))
 
+    # Add or update headers of required files.
+    # Return 1 if files were added or updated, and return 0 if no files were altered.
+    return_code = check_exists(changed_headers, parser, values, proj, missing_headers, 0)
+
+    # Unlink default files & remove .reuse and LICENSES folders if empty
+    cleanup(assets, os_git_root)
+
```

Comment:
Interesting... you are always deleting - even though the repo already had this files. I don't think this is expected, right?

Reply:
It is expected because after the hook is done running, the .reuse and LICENSES folders would still be there so we have to clean them up again as long as there are no custom templates or licenses being used

Reply:
You are right. My point is if they have a _custom_ ansys.jinja2 or MIT license, then those would also get deleted
Anyway, it shouldn't be the case. Those shouldn't be kept under version control by the repos. Let's leave it as is

Reply:
That makes sense. I can add something in the README that says if you are using a custom template or license, it can't be named ansys.jinja2 or MIT.txt, otherwise it'll be removed

Reply:
Yep please do so - just in case somebody asks